rohmu/util.py

```diff
@@ -205,10 +213,10 @@ def seek(self, offset: int, whence: int = 0) -> int:
     def truncate(self, size: Optional[int] = None) -> int:
         raise UnsupportedOperation("truncate")
 
-    def write(self, s: Union[bytes, Buffer]) -> int:
+    def write(self, s: Union[bytes, Buffer]) -> int:  # type: ignore
```

Comment:
Why do you need to ignore the type here (and in `writelines`)?

Reply:
`BinaryIO` wants only `bytes` and no `Buffer` as input for `write` since it's basically just `IO[bytes]`, but I believe in some places we actually do want to forward some buffer instead of `bytes` hence why `Buffer` appears here.

I believe that currently there is no way to avoid ignoring some cases because we have some dependencies that want `BinaryIO` and other places that don't want that in their typing. So if we remove `Buffer` here and make this a "proper" `BinaryIO` then you'll find similar comments when we use this class instead.

Reply:
should we go this way with type ignored? OR should I remove the Union of Buffer here? I tried locally, seems mypy didn't complain other places' typing error.

Reply:
in this repo: only 2 upload_concurent_chunk func calling this class: ProgressStream with fd: BinaryIO, seems all right.

Reply:
I believe we don't have other better option here, right?

Reply:
Yeah I don't think we have better options now. Unfortunately this area of typing is a bit of a mess...

Reply:
The comment is not blocking from my side, was just trying to understand the issue. TY @giacomo-alzetta-aiven for breaking it down. The only concern was that it might hide legit issues that can pop up in the future. Maybe even that is overblown? I don't know.